src/main.cpp

```diff
@@ -96,12 +97,14 @@ void main(void)
 void control_thread()
 {
     uint32_t last_call = 0;
-    int wdt_channel = watchdog_register(200);
+    int thread_id_control = sw_watchdog_register(200);
```

Comment:
Thread ID is something different in Zephyr, so I prefer to keep the name `wdt_channel`. In this file it should be enough to  just replace `watchdog_register` with `sw_watchdog_register` and also start the software watchdog in addition to the hardware watchdog.